### PR TITLE
fix: correctly handle even numbers of backslashes before closing string quote

### PR DESCRIPTION
**Pull request by chatman-media**
## Problem

The `skipString` function in `src/function-tokenizer.ts` used a one-character look-behind to decide whether a closing quote is escaped:

```ts
const prev = source.charAt(pos - 1)
if (ch === quote && prev !== '\\') {
  pos++
  return
}
```

This check is incorrect when an **even** number of backslashes precedes the closing quote. For example, the source string `'test\\'` represents the JS value `test\` (a string ending with one literal backslash). Here the source characters are `' t e s t \ \ '`. Because the character immediately before the closing `'` is a backslash, the old code treats the quote as escaped and runs past the end of the string, ultimately throwing:

```
SyntaxError: Parsing parameter list, did not expect EOF token
```

This manifests in practice whenever a class constructor has a parameter with a string default value whose final character is a backslash (e.g. Windows filesystem paths):

```ts
class MyService {
  constructor(dep1, basePath = 'C:\\Windows\\', dep2) {}
}
```

## Fix

Count **all** consecutive backslashes before the quote. An even count means each pair cancels out (literal backslash characters), so the quote **is** the end of the string. An odd count means the last backslash is an escape for the quote, so the string continues.

```ts
if (ch === quote) {
  let backslashCount = 0
  let i = pos - 1
  while (i >= 0 && source.charAt(i) === '\\') {
    backslashCount++
    i--
  }
  if (backslashCount % 2 === 0) {
    pos++
    return
  }
}
```

## Tests

- Added a new tokenizer test (`can skip strings ending with an even number of backslashes`) covering 2-backslash and 4-backslash cases.
- Added a new param-parser bug regression test (`string default ending with even backslashes`) that directly exercises the real-world failure scenario, including a check that single-backslash escaped quotes still work correctly.

All 235 tests pass.

### CODE REVIEW

File: src/__tests__/function-tokenizer.test.ts [hunk 1]
```diff
@@ -57,6 +57,18 @@ describe('tokenizer', () => {
     ).toMatchSnapshot()
   })
 
+  it('can skip strings ending with an even number of backslashes', () => {
+    // An even number of backslashes before the closing quote means the backslashes
+    // escape each other (e.g. `\\` = one literal backslash), so the quote is NOT escaped.
+    // The previous implementation only checked the immediately preceding character,
+    // causing it to incorrectly treat the closing quote as escaped.
```
Comment (jeffijoe) [inline src/__tests__/function-tokenizer.test.ts]: Please remove the new comments from the test files

File: src/function-tokenizer.ts [hunk 2]
```diff
@@ -215,11 +215,21 @@ export function createTokenizer(source: string) {
...
       const ch = source.charAt(pos)
-      const prev = source.charAt(pos - 1)
-      // Checks if the quote was escaped.
-      if (ch === quote && prev !== '\\') {
-        pos++
-        return
+      // Checks if the quote was escaped by counting consecutive preceding backslashes.
+      // An even number of backslashes means they cancel each other out (e.g. `\\` is a
+      // literal backslash), so the quote is NOT escaped. An odd number means the quote
+      // IS escaped (e.g. `\'`).
```
Comment (jeffijoe) [inline src/function-tokenizer.ts]: Wouldn't checking the immediate next character (and consuming it) if it's also a backslash work as well?

File: src/__tests__/function-tokenizer.test.ts (hunk 1 above)
Comment (chatman-media) [reply src/__tests__/function-tokenizer.test.ts]: Done — removed the comments from both test files in 769ea9d.

File: src/function-tokenizer.ts (hunk 2 above)
Comment (chatman-media) [reply src/function-tokenizer.ts]: Yes, that's cleaner — switched to exactly that in 769ea9d: when a backslash is encountered, both it and the following character are consumed (`pos += 2`), so an escaped quote never terminates the scan and a literal backslash can't swallow the closing quote. Behavior is identical (all snapshots unchanged, 235 tests pass).

File: src/function-tokenizer.ts [hunk 3]
```diff
@@ -215,9 +215,14 @@ export function createTokenizer(source: string) {
     pos++
     while (pos < source.length) {
       const ch = source.charAt(pos)
-      const prev = source.charAt(pos - 1)
-      // Checks if the quote was escaped.
-      if (ch === quote && prev !== '\\') {
+      // Consume escape sequences entirely (e.g. `\'`, `\\`) so neither an
+      // escaped quote nor a literal backslash can affect string termination.
+      if (ch === '\\') {
+        pos += 2
```
Comment (jeffijoe) [inline src/function-tokenizer.ts:221]: Might want to add a test where the very last character in the parsed code is a `\` to make sure this doesn't cause an overflow.

Comment (chatman-media) [reply src/function-tokenizer.ts:221]: Added in 4ba482d — two cases: a lone trailing `\` as the very last character, and a dangling `\\` escape at the end. Both terminate cleanly with an EOF token (the `pos < source.length` guard exits once `pos += 2` steps past the end), snapshots confirm no hang/overflow. Full suite still green (237).

**Comment by jeffijoe** [comment]
@chatman-media I'll need you to rebase this to latest `master` before I can merge it. Thanks!

**Comment by chatman-media** [comment]
Done — rebased onto the latest `master` (now a clean 3-commit history on top of `8b45b70`, no merge commit). The diff is just the tokenizer fix + tests/snapshot. `npm ci` and the tokenizer/param-parser suites pass locally. Ready when you are — thanks!

**Comment by jeffijoe** [comment]
Released as v13.0.5, thanks!
